### DISCUSSION TITLE
Technical stack consideration

### DISCUSSION THREAD

**Discussion by GeraudFabien** [dotnet/Microsoft.Maui.Graphics.Controls#22] [Q&A] [answered]
> There is currently support for:
> 
> iOS using CoreGraphics
> macOS using CoreGraphics
> Android using Android Graphics
> Tizen using SkiaSharp
> Soon, support will be added to:
> 
> Linux using SkiaSharp
> UWP using Win2D
> WPF using SharpDX

I don't understand. When i look Skiasharp work everywhere. What are the reason to use (CoreGraphics, Android Graphics, Win2D, SharpDX)?

Also Uno do pixel perfect. Isn't there a way to share a maximum of code with them? I hope to simplify for us to make cross plateform control and to later support wasm.

It's a real question i just want to understand. I'm not saying "you should use". But maybe i should have post it there https://github.com/dotnet/Microsoft.Maui.Graphics

**Comment by jsuarezruiz** [comment, marked as answer]
The idea in this case is that you can render the controls using the native APIs on each platform by default. In this way, no extra library is required (as would be the case with SkiaSharp) and it has no impact on the size of the library. Although it would also be possible to choose if you want to use SkiaSharp for various reasons (knowledge, extend, etc).
